### ISSUE TITLE
JsonRpcClient isn't reusable

### ISSUE THREAD

**Issue by nikurt** [near/near-jsonrpc-client-rs#25]
The `call` method of `JsonRpcClient` consumes `self`. Because of that if I want to make a chain of RPC calls, I have to keep creating new clients for each call.
Use case: get `block` info and then fetch info for 1-4 `chunk`s. Every request needs to have a new client.

**Comment by miraclx** [comment]
Yeah, artifacts of earlier design iterations, I'll clean this up.

Caused by the observation of what happens in this case with `::call(&self, ..)`:

```rust
let later_response = JsonRpcClient::connect("http://localhost:3030")
    .call(methods::status::RpcStatusRequest);
let response = later_response.await;
```

```console
error[E0716]: temporary value dropped while borrowed
  --> src/main.rs:1:22
   |
49 | let later_response = JsonRpcClient::connect("http://localhost:3030")
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary which is freed while still in use
50 |     .call(methods::status::RpcStatusRequest);
   |                                             - temporary value is freed at the end of this statement
51 | let response = later_response.await;
   |                -------------- borrow later used here
   |
   = note: consider using a `let` binding to create a longer lived value
```

However, this is a fairly trivial issue that shouldn't have required `::call()` taking ownership of the client. The caller just would just have to ensure that the client lives longer than its response future. So;

```rust
let client = JsonRpcClient::connect("http://localhost:3030");
let later_status1 = client.call(methods::status::RpcStatusRequest);
let later_status2 = client.call(methods::status::RpcStatusRequest);
let (status1, status1) = join!(later_status1, later_status2);
```

However, here's a temporary fix for your case: since `.clone()`-ing the client isn't necessarily expensive, both clones would just reference the same internal client data.

```rust
let client = JsonRpcClient::connect("https://localhost:3030");
let later_status1 = client.clone().call(methods::status::RpcStatusRequest);
let later_status2 = client.call(methods::status::RpcStatusRequest);
```

**Comment by miraclx** [comment]
Fixed in 471a53be062e0880c6bc5c2721d123da2a9e0c2e, expect this to land in the next crates.io release.

In the meantime, you can, if you choose to, patch the dependency in your manifest to use this revision.

```toml
[patch.crates-io.near-jsonrpc-client]
git = "https://github.com/near/near-jsonrpc-client-rs"
rev = "471a53be062e0880c6bc5c2721d123da2a9e0c2e"
```
